src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspace.cs

```diff
@@ -114,5 +114,19 @@ public PortableExecutableReference CreatePortableExecutableReference(string file
         }
 
         internal abstract string TryGetRuleSetPathForProject(ProjectId projectId);
+
```

Comment:
Do we want this to instead be a public workspace service so it's not tied to just VisualStudioWorkspace? As @tmat and I are beginning to think about moving the workspace in other places too it seems like this should be somewhere else. I assume nothing would actually be workspace specific, and would just end up calling Workspace.TryApplyChanges so there's probably not a reason to tie it specifically.

I could imagine making it a workspace service, or just a public static method somewhere in a lower layer?

Reply:
I like the idea of a service, spitballing design in another comment related to method name as well.